spec.bs

```diff
@@ -3485,6 +3492,58 @@ A <dfn>server auction response</dfn> is a [=struct=] that contains auction resul
   :: A [=map=] whose [=map/keys=] are [=origins=] and whose [=map/values=] are [=lists=] of [=urls=].
 </dl>
 
+<dfn>server auction join candidate</dfn> is a [=struct=] with the following [=struct/items=]:
+<dl dfn-for="server auction join candidate">
+  : <dfn>ad render url hash</dfn>
+  :: A [=SHA-256=] hash of the [=k-anonymity key=] for the ad in the winning bid,
+     as calculated using [=compute the key hash of ad=].
+  : <dfn>ad component render url hashes</dfn>
+  :: A [=list=] of [=SHA-256=] hashes of [=k-anonymity keys=] for each of the ad
+     components in the winning bid, as calculated using
+     [=compute the key hash of component ad=].
+  : <dfn>reporting id hash</dfn>
+  :: A [=SHA-256=] hash of the [=k-anonymity key=] for the reporting ID in the
+     winning bid, as calculated using [=compute the key hash of reporting ID=].
+</dl>
+
+<dfn>server auction ghost winner</dfn> is a [=struct=] with the following [=struct/items=]:
+<dl dfn-for="server auction ghost winner">
+  : <dfn>candidate</dfn>
+  :: [=server auction join candidate=] associated with this winner.
+  : <dfn>interest group owner</dfn>
+  :: An [=origin=]. The non-k-anonymous winning bid's interest group [=interest group/owner=].
+  : <dfn>interest group name</dfn>
+  :: A [=string=]. The non-k-anonymous winning bid's interest group [=interest group/name=].
+  : <dfn>ghost winner bid info</dfn>
+  :: Null or a [=server auction ghost winner bid info=]. Contains information needed for
+     ghost winners in component auctions.
+</dl>
+
+A <dfn>server auction ghost winner bid info</dfn> is a [=struct=] with the following [=struct/items=]:
+<dl dfn-for="server auction ghost winner bid info">
+  : <dfn>ad render url</dfn>
+  :: A. [=URL=]. The [=leading bid info/leading non-k-anon-enforced bid=]'s
+     [=generated bid/ad descriptor=]'s [=ad descriptor/url=] from the auction.
+  : <dfn>ad components</dfn>
+  :: A [=list=] of [=URLs=]. A list of the  [=leading bid info/leading non-k-anon-enforced bid=]'s
+     [=generated bid/ad component descriptors=]'s [=ad descriptor/urls=] from the
+     server auction.
+  : <dfn>modified_bid</dfn>
+  :: [=bid with currency=]. Contains the non-k-anonymous winning bid's [=generated bid/modified bid=]
+     when not null, otherwise the non-k-anonymous winning bid's [=generated bid/bid=].
+  : <dfn>ad metadata</dfn>

```

Comment:
this and the two id fields below all need to be initially null, since they're only set when the response has corresponding fields: https://github.com/privacysandbox/draft-ietf-bidding-and-auction-services/pull/16/files#diff-740039e394c453ba5ca4c6c7cc2d4bec9399b504e2a279c28e1625ab0e87dbe9R1120. Otherwise, their values are unset, instead of null.

Reply:
Done.